store/postgres/src/notification_listener.rs

```diff
@@ -303,29 +303,32 @@ impl JsonNotification {
             // Use the large_notifications row ID as the payload for NOTIFY
             select(pg_notify(channel, &payload_id.to_string())).execute(conn)?;
 
-            // Delete any notifications older than 5 minutes; this serves
-            // as a regular cleanup so the table doesn't grow in size indefinitely
+            // Prune old large_notifications. We want to keep the size of the
+            // table manageable, but there's a lot of latitude in how often
+            // we need to clean up before old notifications slow down
+            // data access.
             //
-            // We make sure that we do not wait for any locks when we do the
-            // cleanup to avoid stalling this transaction. There is not much
-            // danger in this cleanup sometimes leaving rows behind, as long
-            // as they get cleaned up eventually
-
-            // 'delete' locks the table in 'row exclusive' mode; see that we
-            // can get that lock without waiting
-            let lock =
-                diesel::sql_query("lock table large_notifications in row exclusive mode nowait")
-                    .execute(conn);
```

Comment:
Is it safe to get rid of this Postgres-level lock?

Reply:
Yes. When I put that lock there I meant for it to make it so that only one index node at a time would try to clean up. The lock is the same kind that the `delete` statement takes. I didn't read the fine-print well enough though as any number of connections can take a `row exclusive` lock simultaneously, and the lock served no real purpose.

It would be nice to have some sort of singleton worker that can do the cleanup, but doing it no more often than every configured cleanup interval should get us to almost the same place in terms of load/performance.